sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -551,3 +552,30 @@ object RewriteCorrelatedScalarSubquery extends Rule[LogicalPlan] {
       }
   }
 }
+
+/**
+ * This rule rewrites uncorrelated PredicateSubquery expressions such as Exists.
+ * The uncorrelated Exists can be evaluated using a subplan instead of a semi-join.
+ * Also, we can use `limit 1` and `select 1` after the subquery to reduce the result set.
+ * Example:
+ * exists(select b from t where t.a = 2) => exists(select 1 from t where t.a = 2 limit 1)
```

Comment:
Can you give a more complete example query for this rewrite? It is beneficial for readers.

Reply:
Sure!